source/common/common/version.cc

```diff
@@ -21,18 +25,59 @@ const std::string& VersionInfo::revisionStatus() {
 }
 
 const std::string& VersionInfo::version() {
+  CONSTRUCT_ON_FIRST_USE(std::string,
+                         fmt::format("{}/{}/{}/{}/{}", revision(), BUILD_VERSION_NUMBER,
+                                     revisionStatus(), buildType(), sslVersion()));
+}
+
+const envoy::api::v2::core::BuildVersion& VersionInfo::buildVersion() {
+  static envoy::api::v2::core::BuildVersion result = makeBuildVersion();
+  return result;
+}
+
+const std::string& VersionInfo::buildType() {
 #ifdef NDEBUG
-  static const absl::string_view release_type = "RELEASE";
+  static const std::string release_type = "RELEASE";
```

Comment:
Why was this required?

Reply:
I need to make buildType and sslVersion into functions so that unit test can verify version metadata. I made functions return std::string. I can make it to be 'const char*'.